datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -1090,11 +1090,31 @@ impl LogicalPlanBuilder {
         group_expr: impl IntoIterator<Item = impl Into<Expr>>,
         aggr_expr: impl IntoIterator<Item = impl Into<Expr>>,
     ) -> Result<Self> {
-        let group_expr = normalize_cols(group_expr, &self.plan)?;
+        self._aggregate(group_expr, aggr_expr, true)
+    }
+
+    pub fn aggregate_without_implicit_group_by_exprs(
+        self,
+        group_expr: impl IntoIterator<Item = impl Into<Expr>>,
+        aggr_expr: impl IntoIterator<Item = impl Into<Expr>>,
+    ) -> Result<Self> {
+        self._aggregate(group_expr, aggr_expr, false)
+    }
+
+    fn _aggregate(
+        self,
+        group_expr: impl IntoIterator<Item = impl Into<Expr>>,
+        aggr_expr: impl IntoIterator<Item = impl Into<Expr>>,
+        include_implicit_group_by_exprs: bool,
+    ) -> Result<Self> {
+        let mut group_expr = normalize_cols(group_expr, &self.plan)?;
         let aggr_expr = normalize_cols(aggr_expr, &self.plan)?;
 
-        let group_expr =
-            add_group_by_exprs_from_dependencies(group_expr, self.plan.schema())?;
+        if include_implicit_group_by_exprs {
+            group_expr =
+                add_group_by_exprs_from_dependencies(group_expr, self.plan.schema())?;
+        }
```

Comment:
nit: style-wise I'd prefer keeping `group_expr` non-mut and doing something like:
```suggestion
        let group_expr = if include_implicit_group_by_exprs {
            group_expr =
                add_group_by_exprs_from_dependencies(group_expr, self.plan.schema())?;
        } else {
           group_exrp
        };
```

Reply:
Thank you (TIL)! I made the update.